packages/eslint-plugin/src/rules/explicit-member-accessibility.ts

```diff
@@ -151,6 +162,47 @@ export default util.createRule<Options, MessageIds>({
       }
     }
 
+    /**
+     * Creates a fixer that removes a "public" keyword with following spaces
+     */
+    function getUnwantedPublicAccessibilityFixer(
+      node:
+        | TSESTree.MethodDefinition
+        | TSESTree.ClassProperty
+        | TSESTree.TSParameterProperty,
+    ): ReportFixFunction {
+      return function(fixer: RuleFixer): RuleFix {
+        const tokens = sourceCode.getTokens(node);
+        let rangeToRemove: Range;
+        for (let i = 0; i < tokens.length; i++) {
+          const token = tokens[i];
+          if (
+            token.type === AST_TOKEN_TYPES.Keyword &&
+            token.value === 'public'
+          ) {
+            const commensAfterPublicKeyword = sourceCode.getCommentsAfter(
+              token,
+            );
+            if (commensAfterPublicKeyword.length) {
+              // public /* Hi there! */ static foo()
+              // ^^^^^^^
+              rangeToRemove = [
+                token.range[0],
+                commensAfterPublicKeyword[0].range[0],
+              ];
+              break;
+            } else {
+              // public static foo()
+              // ^^^^^^^
+              rangeToRemove = [token.range[0], tokens[i + 1].range[0]];
+              break;
+            }
+          }
+        }
```

Comment:
This looks nice indeed, but:

> // also remove the trailing space which must exist

it not true, as keyword can be followed with a comment as in tests:

`contructor(public/* Hi there */ readonly foo)`

so we anyway need the next token. And since we have the range for the following token, I believe the rule should delete all the spaces between them, not just one.

Reply:
Mmm true.
I personally wouldn't have worried about saving the comment in a weird place, but if you've got the code written already, sure.